### PR TITLE
Endpoint to assign a job to user and to retrieve assigned jobs

### LINKED ISSUE #11: Job assignment
- Human App sends a request to the Exchange Oracle to assign the chosen job to the user (job identifier + user identifier).
- Human App displays a comprehensive list of jobs assigned to the user.
- Each job has a link to the respective annotation tool for completion.

https://tech-docs.humanprotocol.org/architecture/exchange-oracle#request-to-get-a-job-assigned-to-a-specific-user

https://tech-docs.humanprotocol.org/architecture/exchange-oracle#allows-retrieving-a-list-of-jobs-assigned-to-a-specific-worker

### PR DESCRIPTION
**Pull request by MWBlocky**
## Description

This Pull Request adds the jobs asignment endpoints.

## Summary of changes

- Added job assignment module.
- Added unit and e2e tests.

## Related issues
Fixes #11

### CODE REVIEW

File: packages/apps/human-app/server/src/modules/job-assignment/interfaces/job-assignment.interface.ts [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+  @AutoMap()
+  job_type: string;
+  @AutoMap()
+  status: StatusEnum;
+  @AutoMap()
+  page_size: number;
+  @AutoMap()
+  page: number;
+  @AutoMap()
+  sort: 'ASC' | 'DESC';
```
Comment (BartoszSolkaBD) [inline packages/apps/human-app/server/src/modules/job-assignment/interfaces/job-assignment.interface.ts:115]: maybe create an enum type?

File: packages/apps/human-app/server/src/modules/job-assignment/job-assignment.service.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+@Injectable()
+export class JobAssignmentService {
+  constructor(
+    public httpService: HttpService,
+    @InjectMapper() private readonly mapper: Mapper,
+  ) {}
+
+  async processJobAssignment(
+    url: string,
```
Comment (BartoszSolkaBD) [inline packages/apps/human-app/server/src/modules/job-assignment/job-assignment.service.ts:23]: url can be part of the command

File: packages/apps/human-app/server/src/modules/job-assignment/job-assignment.controller.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+import { InjectMapper } from '@automapper/nestjs';
+import { Mapper } from '@automapper/core';
+import { JobAssignmentService } from './job-assignment.service';
+import {
+  JobAssignmentCommand,
+  JobAssignmentDto,
+  JobsAssignmentParamsDto,
+  JobAssignmentResponse,
+  JobsAssignmentParamsCommand,
+  JobsAssignmentResponse,
```
Comment (macnablocky) [inline packages/apps/human-app/server/src/modules/job-assignment/job-assignment.controller.ts:21]: why two times

File: packages/apps/human-app/server/src/modules/job-assignment/job-assignment.service.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+      const response = await lastValueFrom(this.httpService.request(options));
+      return response.data;
+    } catch (error) {
+      if (error.response) {
+        throw new HttpException(error.response.data, error.response.status);
+      } else {
+        throw new HttpException(
+          'Internal Server Error',
+          HttpStatus.INTERNAL_SERVER_ERROR,
+        );
```
Comment (macnablocky) [inline packages/apps/human-app/server/src/modules/job-assignment/job-assignment.service.ts:46]: merge with main and use exception filter

File: packages/apps/human-app/server/src/modules/job-assignment/job-assignment.service.ts [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
```
Comment (macnablocky) [inline packages/apps/human-app/server/src/modules/job-assignment/job-assignment.service.ts:75]: merge with main and use exception filter

File: packages/apps/human-app/server/src/modules/job-assignment/job-assignment.service.ts [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+  }
+
+  async processGettingAssignedJobs(
+    url: string,
+    jobsAssignmentParamsCommand: JobsAssignmentParamsCommand,
+  ): Promise<JobsAssignmentResponse> {
+    const jobsAssignmentParamsData = this.mapper.map(
+      jobsAssignmentParamsCommand,
+      JobsAssignmentParamsCommand,
+      JobsAssignmentParamsData,
```
Comment (macnablocky) [inline packages/apps/human-app/server/src/modules/job-assignment/job-assignment.service.ts:58]: I don't understand usage of this: 
 In my opinion - command should be ready to send request, this is datatype used in the service layer, data should be datatype returned to the controller - so I would remove this additional mapping, and changed returned values from ...Response to ...Data

File: packages/apps/human-app/server/src/modules/job-assignment/job-assignment.service.ts [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+      } else {
+        throw new HttpException(
+          'Internal Server Error',
+          HttpStatus.INTERNAL_SERVER_ERROR,
+        );
+      }
+    }
+  }
+
+  async processGettingAssignedJobs(
```
Comment (macnablocky) [inline packages/apps/human-app/server/src/modules/job-assignment/job-assignment.service.ts:51]: get/fetchAssignedJobs

File: packages/apps/human-app/server/src/modules/job-assignment/job-assignment.service.ts (hunk 6 above)
Comment (MWBlocky) [reply packages/apps/human-app/server/src/modules/job-assignment/job-assignment.service.ts:58]: `~Data` type is used for external API's for request purposes and `~Response` data type for their returns (look at `user-operator` and `user-worker` modules)

File: packages/apps/human-app/server/src/modules/job-assignment/job-assignment.service.ts (hunk 2 above)
Comment (MWBlocky) [reply packages/apps/human-app/server/src/modules/job-assignment/job-assignment.service.ts]: Fixed in: 4ba85d0b

File: packages/apps/human-app/server/src/modules/job-assignment/interfaces/job-assignment.interface.ts [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+  @AutoMap()
+  sort_field: 'chain_id' | 'job_type' | 'reward_amount' | 'created_at';
```
[the remark this answers is not in the document]
Comment (MWBlocky) [reply packages/apps/human-app/server/src/modules/job-assignment/interfaces/job-assignment.interface.ts]: Fixed in: 4ba85d0b

File: packages/apps/human-app/server/src/modules/job-assignment/interfaces/job-assignment.interface.ts (hunk 1 above)
Comment (MWBlocky) [reply packages/apps/human-app/server/src/modules/job-assignment/interfaces/job-assignment.interface.ts]: Fixed in: 4ba85d0b

File: packages/apps/human-app/server/src/modules/job-assignment/job-assignment.service.ts (hunk 4 above)
Comment (MWBlocky) [reply packages/apps/human-app/server/src/modules/job-assignment/job-assignment.service.ts]: Fixed in: 4ba85d0b

File: packages/apps/human-app/server/src/modules/job-assignment/job-assignment.controller.ts [hunk 9]
```diff
@@ -13,10 +12,10 @@ import { InjectMapper } from '@automapper/nestjs';
 import { Mapper } from '@automapper/core';
 import { JobAssignmentService } from './job-assignment.service';
 import {
-  JobAssignmentCommand,
   JobAssignmentDto,
-  JobsAssignmentParamsDto,
+  JobAssignmentCommand,
   JobAssignmentResponse,
+  JobsAssignmentParamsDto,
```
Comment (macnablocky) [inline packages/apps/human-app/server/src/modules/job-assignment/job-assignment.controller.ts:18]: imo this naming convention has two flaws: 
1. Is to similar by usecase
2. JobsAssignment... don't tell about real usecase - maybe: JobsFetch Dto/Params/Command/Response,

**Review by macnablocky — APPROVED** [review]

File: packages/apps/human-app/server/src/modules/job-assignment/job-assignment.controller.ts (hunk 9 above)
Comment (MWBlocky) [reply packages/apps/human-app/server/src/modules/job-assignment/job-assignment.controller.ts]: Fixed in: a1b51198

File: packages/apps/human-app/server/src/modules/job-assignment/spec/job-assignment.fixtures.ts [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+  JobsFetchResponseItem,
+} from '../interfaces/job-assignment.interface';
+import {
+  SortField,
+  SortOrder,
+  StatusEnum,
+} from '../../../common/enums/job-assignment';
+
+export const jobAssignmentDtoFixture: JobAssignmentDto = {
+  exchange_oracle_url: 'test_url',
```
Comment (BartoszSolkaBD) [inline packages/apps/human-app/server/src/modules/job-assignment/spec/job-assignment.fixtures.ts:19]: test with an example URL (https://www.example.com/api)

**Review by BartoszSolkaBD — APPROVED** [review]

File: packages/apps/human-app/server/src/modules/job-assignment/spec/job-assignment.fixtures.ts (hunk 10 above)
Comment (MWBlocky) [reply packages/apps/human-app/server/src/modules/job-assignment/spec/job-assignment.fixtures.ts]: Fixed in: 051f0adc

**Pull request merged**